modules/javafx.graphics/src/main/java/com/sun/javafx/stage/WindowRelocator.java

```diff
@@ -0,0 +1,327 @@
+     */
+    public static WindowLocationAlgorithm newRelocationAlgorithm(Screen userScreen,
+                                                                 AnchorPoint screenAnchor,
+                                                                 Insets screenPadding,
+                                                                 AnchorPoint stageAnchor,
+                                                                 AnchorPolicy anchorPolicy) {
+        Objects.requireNonNull(screenAnchor, "screenAnchor cannot be null");
+        Objects.requireNonNull(screenPadding, "screenPadding cannot be null");
+        Objects.requireNonNull(stageAnchor, "stageAnchor cannot be null");
+        Objects.requireNonNull(anchorPolicy, "anchorPolicy cannot be null");
+
+        return (windowScreen, windowWidth, windowHeight) -> {
+            double screenX, screenY;
+            double gravityX, gravityY;
+            Screen currentScreen = Objects.requireNonNullElse(userScreen, windowScreen);
+            Rectangle2D currentBounds = Utils.hasFullScreenStage(currentScreen)
+                ? currentScreen.getBounds()
+                : currentScreen.getVisualBounds();
+
+            // Compute the absolute coordinates of the screen anchor.
+            // If the screen anchor is specified in proportional coordinates, it is proportional to the complete
+            // screen bounds when a full-screen stage is showing on the screen, and the visual bounds otherwise.
+            if (screenAnchor.isProportional()) {
+                screenX = currentBounds.getMinX() + screenAnchor.getX() * currentBounds.getWidth();
+                screenY = currentBounds.getMinY() + screenAnchor.getY() * currentBounds.getHeight();
+            } else {
+                screenX = currentBounds.getMinX() + screenAnchor.getX();
+                screenY = currentBounds.getMinY() + screenAnchor.getY();
+            }
+
+            // The absolute screen anchor might be on a different screen than the current window, so we
+            // need to recompute the actual screen and its bounds (complete when full-screen stage showing,
+            // visual otherwise).
+            Screen targetScreen = Utils.getScreenForPoint(screenX, screenY);
+            Rectangle2D screenBounds = Utils.hasFullScreenStage(targetScreen)
+                ? targetScreen.getBounds()
+                : targetScreen.getVisualBounds();
+
+            Point2D location = computeAdjustedLocation(
+                screenX, screenY,
+                windowWidth, windowHeight,
+                stageAnchor, anchorPolicy,
+                screenBounds, screenPadding);
+
+            if (stageAnchor.isProportional()) {
+                gravityX = stageAnchor.getX();
+                gravityY = stageAnchor.getY();
+            } else {
+                gravityX = stageAnchor.getX() / windowWidth;
+                gravityY = stageAnchor.getY() / windowHeight;
```

Comment:
It might be a good idea to explain what "gravity" is, maybe in `ComputedLocation` or `WindowLocationAlgorithm`.

Here, for example, L101 tells us the gravity is measured in pixels, while L104 indicates the gravity is dimensionless.